### DISCUSSION TITLE
Which hook to fetch API data with recoil

### DISCUSSION THREAD

**Discussion by ivanjeremic** [facebookexperimental/Recoil#997] [Q&A]
The docs say for selectors `Please keep in mind that selectors represent "idempotent" functions: For a given set of inputs they should always produce the same results (at least for the lifetime of the application).`

Not sure I understand it but it sounds to me selectors need to have some atom state they depend on and the data returned by the selector is not supposed to change, so for me, it sounds like `selectors & slectorFamily` is not something I would use to fetch for example some blog posts from an rest api or can I?

**Comment by BenjaBobs** [comment]
You can use selectors/selectorFamilies to fetch data such as blog posts. There's an example of how to do that here: https://github.com/facebookexperimental/Recoil/discussions/996
